roles/ssl/tasks/idp_certs.yml

```diff
@@ -0,0 +1,52 @@
+---
+- name: Copy IDP Cert to Node
+  copy:
+    src: "{{idp_cert_path}}"
+    dest: "{{idp_cert_dest}}"
+  diff: "{{ not mask_sensitive_diff|bool }}"
+
+- name: Check if Cert already imported
+  shell: |
+    keytool -list -keystore {{truststore_path}} \
+    -storetype pkcs12 -alias {{alias}} -storepass {{truststore_storepass}}
+  ignore_errors: true
+  register: cert_imported
+  no_log: "{{mask_secrets|bool}}"
+
+- name: Import IdP certificate to Truststore
+  shell: |
+    keytool -noprompt -keystore {{truststore_path}} \
+      -storetype pkcs12 \
+      -alias {{alias}} \
+      -import -file {{idp_cert_dest}} \
+      -storepass {{truststore_storepass}}
+  when:
+    - cert_imported.rc == 1
+  no_log: "{{mask_secrets|bool}}"
+
+- name: Check if Cert already imported
+  shell: |
+    keytool -list -keystore {{bcfks_truststore_path}} \
+    -storetype BCFKS -alias {{alias}} \
+    -storepass {{truststore_storepass}} \
+    -providerclass org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+    -providerpath {{ (binary_base_path, 'share/java/kafka/bc-fips-*.jar') | path_join }}
+  ignore_errors: true
+  register: cert_imported
+  when:
+    - fips_enabled|bool
+  no_log: "{{mask_secrets|bool}}"
+
+- name: Import IdP certificate to BCFKS Truststore
+  shell: |
+    keytool -noprompt -keystore {{bcfks_truststore_path}} \
+      -storetype BCFKS \
+      -alias {{alias}} \
+      -import -file {{idp_cert_dest}} \
+      -storepass {{truststore_storepass}} \
+      -providerclass org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+      -providerpath {{ (binary_base_path, 'share/java/kafka/bc-fips-*.jar') | path_join }}
+  when:
+    - create_bouncy_castle_keystore|bool
+    - cert_imported.rc == 1
+  no_log: "{{mask_secrets|bool}}"
```

Comment:
There is task `Delete temporary keys/certs when keystore and trustore is provided` in each of component's roles. So I was wondering if something similar should be done here here after the setup

Reply:
yeah it can be a nice addition, but will need to weigh the pros and cons of deleting the certs for re-runs. Didn't add any delete task since there is no con if the certs are lying on managed nodes but yes something we can consider